### DISCUSSION TITLE
Subscription with multiple physical products and delayed order

### DISCUSSION THREAD

**Discussion by VojtaSim** [swellstores/community#227] [Help] [answered]
Hello, I'd like to ask what's the difference between product with type `subscription` and product with type `bundle`.
I'm trying to create single subscription with multiple physical products via cart (I should be able to add the products to cart)
 - each user can have different combination of products (the bundle is created via API before checkout)
 - I have webhook in place to update the bundle price when bundled products prices change
 
[as per docs](https://developers.swell.is/backend-api/subscription-plans/create-a-plan-with-physical-items) it's suggested to create product with type `subscription`. 
However this product is then displayed in admin among other products and its detail doesn't display much not even the bundle items... What is its purpose then when `bundle` seems better?

Also when I created a subscription with bundled products (either `subscription` or `bundle`) it set first order to be created in a month, not immediately (subscription is active but there's no invoice or order)...?

<img width="909" alt="image" src="https://user-images.githubusercontent.com/4440970/233308150-ee5fa3ef-8829-4bfa-ad10-f9ce73b0e144.png">

**Comment by greph** [comment, marked as answer]
Hey @VojtaSim , a key difference is that a 'subscription' is a product purchase option (recurring, one-time, etc), while the 'bundle' is a product type (physical, digital, bundle, etc)

The subscription purchase option allows you to create plans with different intervals, prices, and trial periods (even split fulfillment). When customers subscribe, they are billed regularly according to the plan they choose (from pre-set options), and orders are auto-generated. On the other hand, a product with type "bundle" is a collection of multiple products that are sold together as a single unit, typically at a discounted price. 

A single subscription plan could be created where customers update or swap out products during its lifecycle, but in this case it sounds like you want to create the bundle-type products on-the-fly. For this reason the 'subscription' purchase option will need to be set on each new bundle product created, potentially with the dynamic pricing you alluded to. 

Regarding the first order getting created in a month for the subscription, keep in mind that an initial order was created when the customer completed checkout. The subscription plan then kicks into action and the next order is set to generate at the next billing cycle (in a month).
